### DISCUSSION TITLE
Why can't renovate get changelogs without a GitHub token?

### DISCUSSION THREAD

**Discussion by benjimin** [renovatebot/renovate#29589] [Request Help] [answered]
### What would you like help with?

I would like help with my configuration

### How are you running Renovate?

Self-hosted

### If you're self-hosting Renovate, tell us which platform (GitHub, GitLab, etc) and which version of Renovate.

Bitbucket, latest docker image

### Please tell us more about your question or problem

Is it compulsory to set `GITHUB_COM_TOKEN` for full renovate functionality?

The docs state: 

> The GitHub.com token as an environment variable is needed to fetch changelogs that are usually hosted on github.com. You don't need to add it if you are already running the bot against github.com, but you do need to add it if you're using GitHub Enterprise Server, GitLab, Azure DevOps, or Bitbucket.

Without this, renovate is creating PRs containing the warning:

> Release Notes retrieval for this PR were skipped because no github.com credentials were available.

Why can't renovate fetch changelogs anonymously, unauthenticated?

I'd rather not deploy renovate (for my team/org) with a token that is associated to a specific individual's GitHub account, is there any alternative?

**Comment by rarkins** [comment, marked as answer]
Because GitHub GraphQL needs mandatory authentication. Also you hit rate limits far too quickly even if anonymous access were possible.

**Reply by HonkingGoose** [reply]
@benjimin you may also want to read the [Renovate docs, Renovate and changelogs](https://docs.renovatebot.com/key-concepts/changelogs/) page. It explains how Renovate "sees" changelogs, and what conditions need to be met for it to find changelogs.
